api/src/main/java/io/opentelemetry/api/trace/TraceStateBuilder.java

```diff
@@ -50,20 +54,21 @@ public TraceStateBuilder set(String key, String value) {
       return this;
     }
     // Initially create the Entry to validate input.
-    TraceState.Entry entry = TraceState.Entry.create(key, value);
     if (entries == null) {
       // Copy entries from the parent.
       entries = new ArrayList<>(parent.getEntries());
     }
-    for (int i = 0; i < entries.size(); i++) {
-      if (entries.get(i).getKey().equals(entry.getKey())) {
+    for (int i = 0; i < entries.size(); i += 2) {
+      if (entries.get(i).equals(key)) {
+        entries.remove(i);
         entries.remove(i);
         // Exit now because the entries list cannot contain duplicates.
         break;
       }
     }
     // Inserts the element at the front of this list.
-    entries.add(0, entry);
+    entries.add(0, key);
```

Comment:
Inserting at the head of ArrayList looks like lots of overhead for the standard case of having no dupes, etc. Can we instead build up in opposite order and `Collections.reverse` in `build`? We'd need to reverse when copying from `parent.getEntries()` but I guess that's just a couple more lines of code.

Reply:
we could also consider just waiting until the final build to de-dupe. There are options here, for sure. Luckily, it's all internal, so we can make it better whenever.

Reply:
we could put the working data into a `LinkedList` for easy insert efficiency, as well. there are definitely options here. Not sure how much I want to take on this PR, though.